com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -1265,6 +1287,76 @@ static void CheckAndApplyDebugSettings(ref RenderingData renderingData)
             }
         }
 
+        /// <summary>
+        /// Returns the best supported image upscaling filter based on the provided upscaling filter selection
+        /// </summary>
+        /// <param name="imageSize">Size of the final image</param>
+        /// <param name="renderScale">Scale being applied to the final image size</param>
+        /// <param name="selection">Upscaling filter selected by the user</param>
+        /// <returns>Either the original filter provided, or the best replacement available</returns>
+        static ImageUpscalingFilter ResolveUpscalingFilterSelection(Vector2 imageSize, float renderScale, UpscalingFilterSelection selection)
+        {
+            // By default we just use linear filtering since it's the most compatible choice
+            ImageUpscalingFilter filter = ImageUpscalingFilter.Linear;
+
+            // Fall back to the automatic filter if FSR was selected, but isn't supported on the current platform
+            if ((selection == UpscalingFilterSelection.FSR) && !FSRUtils.IsSupported())
+            {
+                selection = UpscalingFilterSelection.Auto;
+            }
+
+            switch (selection)
+            {
+                case UpscalingFilterSelection.Auto:
+                {
+                    // The user selected "auto" for their upscaling filter so we should attempt to choose the best filter
+                    // for the current situation. When the current resolution and render scale are compatible with integer
+                    // scaling we use the point sampling filter. Otherwise we just use the default filter (linear).
+                    float pixelScale = (1.0f / renderScale);
+                    bool isIntegerScale = ((pixelScale - Mathf.Floor(pixelScale)) == 0.0f);
+
+                    if (isIntegerScale)
+                    {
+                        float widthScale = (imageSize.x / pixelScale);
+                        float heightScale = (imageSize.y / pixelScale);
+
+                        bool isImageCompatible = (((widthScale - Mathf.Floor(widthScale)) == 0.0f) &&
+                                                  ((heightScale - Mathf.Floor(heightScale)) == 0.0f));
```

Comment:
There's a risk of float imprecision using `a - b == 0.0f`, prefer using `Math.Abs(a - b) < epsilon`